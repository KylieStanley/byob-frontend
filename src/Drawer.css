.drawer {
  height: 100%;
  width: 100%;
  z-index: 1;
  background-color: #fff;
  position: absolute;
  top: 100%;
  transition: all 1s;
}

.drawer-hide {

}

.drawer-view {
  position: absolute;
  top: 160px;
}

.tab-container {
  width: 100%;
  height: 40px;
  display: flex;
}

.tab {
  align-items: center;
  justify-content: center;
  display: flex;
  width: 100px;
  height: 40px;
  margin: 0 auto;
  transition: background-color 1s, 
              border-top-left-radius 1s 1.5s,
              border-top-right-radius 1s 1.5s;
}

.tab-view {
  background-color: #fff;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.tab-hide {
  background-color: gray;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

.mock-sidebar {
  transition: width 1s 1.5s,
              height 1s,
              background-color 1s;
}

.mock-sidebar-hide {
  width: 300px;
  height: 40px;
  background-color: #989898;
}

.mock-sidebar-view {
  width: 0;
  background-color: transparent;
}

.drawer-view-btn:hover {
  cursor: pointer;
}

