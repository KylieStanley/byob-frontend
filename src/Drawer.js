import React, { Component } from 'react';
import './Drawer.css'
import Form from './Form'
import ExampleContainer from './ExampleContainer'


export default class Drawer extends Component {
  constructor() {
    super()
    this.state = {
      showDrawer: false,
      example: {}
    }
  }

  handleResult = (example) => {
    this.setState({ example })
  }

  changeDrawerDisplayState = (event) => {
    event.preventDefault()
    this.setState({ showDrawer: !this.state.showDrawer })
  }

  changeDrawerClass = () => {
    if (this.state.showDrawer) {
      return "drawer drawer-view"
    } else {
      return "drawer drawer-hide"
    }
  }

  changeMockSideClass = () => {
    if (this.state.showDrawer) {
      return "mock-sidebar mock-sidebar-view"
    } else {
      return "mock-sidebar mock-sidebar-hide"
    }
  }

  changeTabClass = () => {
    if (this.state.showDrawer) {
      return "tab tab-view"
    } else {
      return "tab tab-hide"
    }
  }

  render() {
    return (
      <div className={ this.changeDrawerClass() }>
        <div className="tab-container">
          <div className={ this.changeMockSideClass() }></div>
          <div className={ this.changeTabClass() }>
            <i onClick={ this.changeDrawerDisplayState } className="fas fa-bars drawer-view-btn"></i>
          </div>
        </div>

        <div>
          <Form handleResult={ this.handleResult } />
          <ExampleContainer example={ this.state.example } />
        </div>
      </div>
    )  
  }  
}

