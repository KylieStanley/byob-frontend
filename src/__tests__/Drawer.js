import React from 'react'
import { shallow } from 'enzyme'
import Drawer from '../Drawer'


describe('Drawer', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Drawer />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})