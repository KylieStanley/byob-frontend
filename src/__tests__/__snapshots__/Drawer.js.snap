// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Drawer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tab-container"
        >
          <div
            className="mock-sidebar mock-sidebar-hide"
          />
          <div
            className="tab tab-hide"
          >
            <i
              className="fas fa-bars drawer-view-btn"
              onClick={[Function]}
            />
          </div>
        </div>,
        <div>
          <Form
            handleResult={[Function]}
          />
          <ExampleContainer
            example={Object {}}
          />
        </div>,
      ],
      "className": "drawer drawer-hide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mock-sidebar mock-sidebar-hide"
            />,
            <div
              className="tab tab-hide"
            >
              <i
                className="fas fa-bars drawer-view-btn"
                onClick={[Function]}
              />
            </div>,
          ],
          "className": "tab-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "mock-sidebar mock-sidebar-hide",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-bars drawer-view-btn"
                onClick={[Function]}
              />,
              "className": "tab tab-hide",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-bars drawer-view-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Form
              handleResult={[Function]}
            />,
            <ExampleContainer
              example={Object {}}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleResult": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "example": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tab-container"
          >
            <div
              className="mock-sidebar mock-sidebar-hide"
            />
            <div
              className="tab tab-hide"
            >
              <i
                className="fas fa-bars drawer-view-btn"
                onClick={[Function]}
              />
            </div>
          </div>,
          <div>
            <Form
              handleResult={[Function]}
            />
            <ExampleContainer
              example={Object {}}
            />
          </div>,
        ],
        "className": "drawer drawer-hide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mock-sidebar mock-sidebar-hide"
              />,
              <div
                className="tab tab-hide"
              >
                <i
                  className="fas fa-bars drawer-view-btn"
                  onClick={[Function]}
                />
              </div>,
            ],
            "className": "tab-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "mock-sidebar mock-sidebar-hide",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-bars drawer-view-btn"
                  onClick={[Function]}
                />,
                "className": "tab tab-hide",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-bars drawer-view-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Form
                handleResult={[Function]}
              />,
              <ExampleContainer
                example={Object {}}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleResult": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "example": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
